tmp_bench/benchmark.py

```diff
@@ -0,0 +1,195 @@
+# %%
+from sklearn import datasets
+from sklearn.linear_model import LogisticRegression
+from sklearn.preprocessing import StandardScaler
+from sklearn import metrics
+
+import numpy as np
+import time
+import functools
+from joblib import Memory
+import pandas as pd
+import argparse
+import streamlit as st
+import altair as alt
+
+parser = argparse.ArgumentParser(
+    description="Determine whether to save/load the dataframe."
+)
+parser.add_argument(
+    "--save",
+    type=str,
+    default="",
+    metavar="s",
+    help="Saves the dataframe to the path if provided",
+)
+parser.add_argument(
+    "--load",
+    type=str,
+    default="",
+    metavar="l",
+    help="Loads the dataframe at the path if provided",
+)
+parser.add_argument(
+    "--val",
+    action="store_true",
+    help="Validates convergence of trust-ncg against lbfgs",
+)
+
+args = parser.parse_args()
+
+
+def main() -> pd.DataFrame:
+    # %%
+    m = Memory(cachedir=".")
+
+    # %%
+    """
+    TOY_DSETS = {
+        'iris':functools.partial(datasets.load_iris,return_X_y=True),
+        'wine':functools.partial(datasets.load_wine,return_X_y=True),
+        'breast_cancer':functools.partial(datasets.load_breast_cancer,return_X_y=True),
+        'news':functools.partial(datasets.fetch_20newsgroups_vectorized,return_X_y=True)
+        }
+    """
+    TOY_DSETS = {}
+    BASE_SYNTH_DSETS = {
+        "base": datasets.make_classification,
+        "dense": functools.partial(datasets.make_classification, n_informative=15),
+        "redundant": functools.partial(datasets.make_classification, n_redundant=15),
+        "sparse": functools.partial(datasets.make_classification, n_features=200),
```

Comment:
For future reference, best not to use "sparse" in this context, because it will be confused with a SciPy sparse matrix.